plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/components/upgrade-product-image-gallery/index.tsx

```diff
@@ -0,0 +1,73 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState, useEffect } from '@wordpress/element';
+import { createBlock, BlockInstance } from '@wordpress/blocks';
+import { dispatch, useSelect } from '@wordpress/data';
+import { UpgradeDowngradeNotice } from '@woocommerce/editor-components/upgrade-downgrade-notice';
+import { findBlock } from '@woocommerce/utils';
+
+const upgradeToBlockifiedProductGallery = (
+	productImageGalleryBlockClientId: string
+) => {
+	const newBlock = createBlock( 'woocommerce/product-gallery' );
+
+	dispatch( 'core/block-editor' ).replaceBlock(
+		productImageGalleryBlockClientId,
+		newBlock
+	);
+
+	return true;
```

Comment:
I'm a bit confused here, sorry for not following. Isn't this duplicate/but worse code than [this](https://github.com/woocommerce/woocommerce/pull/61201/files#diff-7e81402edd8e7393f762201a096df7559c6b1f04c4d6a060c44a5f6471b1b831R16)?

Reply:
It's similar, but the one you linked is triggered from the _Product Image Gallery_ notice while this one is triggered from the _Add to Cart + Options_ notice. They are slightly different because in the _Product Image Gallery_ block, we get the `clientId` for free, while in the _Add to Cart + Options_ block we need to find the block and then pass the `clientId` to the upgrade function.